### ISSUE TITLE
Sesame Touch becomes idle after the finger mode is changed from "add" to "verify"

### ISSUE THREAD

**Issue by js4jiang5** [CANDY-HOUSE/SesameSDK_ESP32_with_DemoApp#8]
Update:
The solution no longer works after firmware update.
======
I can add the fingerprints and cards by setting the finger and card mode to "add" mode. After the add is done, the mode is changed back to "verify". The card is changed to "verify" mode immediately with no problem. However, the finger becomes idle with no action at all when I touch it with finger. It's neither in "add" mode nor "verify" mode. The only solution is to unplug and plug ESP32 to make the finger mode back to normal "verify" mode. Any idea how to fix it?

**Comment by MotoZe** [comment]
It seems like there's an issue with the ESP32 not properly toggling between the card verification and enrollment modes of ssm_touch. After you set it to the enrollment mode and register the cards, you need to manually switch it back to verification mode for it to start recognizing again.

For a detailed implementation guide, please refer to this document:
https://github.com/CANDY-HOUSE/Sesame_BluetoothAPI_document/blob/master/SesameOS3/item_details_touch/114_card_mode_set.md

Feel free to reach out if you have more questions or need further assistance!

**Comment by js4jiang5** [comment]
> It seems like there's an issue with the ESP32 not properly toggling between the card verification and enrollment modes of ssm_touch. After you set it to the enrollment mode and register the cards, you need to manually switch it back to verification mode for it to start recognizing again.
> 
> For a detailed implementation guide, please refer to this document: https://github.com/CANDY-HOUSE/Sesame_BluetoothAPI_document/blob/master/SesameOS3/item_details_touch/114_card_mode_set.md
> 
> Feel free to reach out if you have more questions or need further assistance!

You misunderstood. The card is normal. It's Finger that behaves weird.

**Comment by MotoZe** [comment]
I apologize for the confusion earlier, and thank you for your patience. Regarding the fingerprint issue, the method to switch between modes is indeed similar to that of the card. You need to manually send specific commands to switch to either "verify" or "enroll" mode. I suggest you refer to the following document, which provides detailed steps and command settings to help resolve the issue you're experiencing:

[Fingerprint Mode Setting Guide](https://github.com/CANDY-HOUSE/Sesame_BluetoothAPI_document/blob/master/SesameOS3/item_details_touch/122_finger_mode_set.md)

**Comment by js4jiang5** [comment]
> I apologize for the confusion earlier, and thank you for your patience. Regarding the fingerprint issue, the method to switch between modes is indeed similar to that of the card. You need to manually send specific commands to switch to either "verify" or "enroll" mode. I suggest you refer to the following document, which provides detailed steps and command settings to help resolve the issue you're experiencing:
> 
> [Fingerprint Mode Setting Guide](https://github.com/CANDY-HOUSE/Sesame_BluetoothAPI_document/blob/master/SesameOS3/item_details_touch/122_finger_mode_set.md)

There's something wrong with "finger mode set". The response is 1 byte less than "card mode set". I've found a way to bring it back to verify mode, though.  Case closed for me.

**Comment by MotoZe** [comment]
Thank you for the update and for sharing your solution! It's great to hear that you've resolved the issue.

**Comment by iammicloud** [comment]
> # Update:
> The solution no longer works after firmware update.

@js4jiang5 
Have you fixed it? I installed your solution and integrated sesame 5 and touch into HA. It worked for months perfectly. After firmware upgrade, I got an error message "Service: CHPrivateAPIClient: Status code 403". HA still can lock/unlock sesame 5, but the battery dropped from 100 to 0% within 1 month after new batteries were replaced. Any idea?
